src/osdc/Objecter.cc

```diff
@@ -2812,9 +2809,9 @@ int Objecter::_calc_target(op_target_t *t, bool any_change)
     } else {
       int osd;
       bool read = is_read && !is_write;
-      if (t->use_osd_epoch) {
+      if (t->forced_osd != -1) {
 	assert(read);
-	osd = osdmap->is_up(t->osd) ? t->osd : -1;
+	osd = osdmap->is_up(t->forced_osd) ? t->forced_osd : -1;
```

Comment:
hmm, even if the osd is up, it may no longer hold the pg for the repair read op - in which case it'll drop the op.

I suppose we could check if it's in acting here, and if not cancel the repair read request with ERANGE on the client side... a bit ugly to add that to the client though...